pandas/core/frame.py

```diff
@@ -1305,11 +1309,17 @@ def to_excel(self, excel_writer, sheet_name='Sheet1', na_rep='',
         """
         from pandas.io.excel import ExcelWriter
         if self.columns.nlevels > 1:
-            raise NotImplementedError("Writing as Excel with a MultiIndex is "
```

Comment:
I'm not exactly sure what you're looking for.  Just the 'raise NotImpl...' moved into the ExcelFormatter __init__?  or where?